socs/agents/acu/agent.py

```diff
@@ -2702,15 +2751,228 @@ def _get_sunsafe_moves(self, target_az, target_el, zero_legs_ok=True):
         if self.sun.check_trajectory([target_az], [target_el])['sun_time'] <= 0:
             return None, 'Requested target position is not Sun-Safe.'
 
-        moves = self.sun.analyze_paths(az, el, target_az, target_el)
+        moves = self.sun.analyze_paths(az0, el0, target_az, target_el)
         move, decisions = self.sun.select_move(moves)
         if move is None:
             return None, 'No Sun-Safe moves could be identified!'
 
         legs = list(move['moves'].nodes)
-        if len(legs) == 1 and not zero_legs_ok:
-            return legs, None
-        return legs[1:], None
+        if len(legs) == 1:
+            # Pad to two entries.
+            return [legs[0], legs[0]], None
+        return legs, None
+
+    #
+    # HWP State Safety
+    #
+
+    @inlineCallbacks
+    def monitor_hwp(self, session, params):
+        """monitor_hwp()
+
+        **Process** - Monitors the state of a HWP, by querying a
+        HWPSupervisor's ``monitor`` Process session data.  Assesses
+        what motions are permitted, given the HWP state.
+
+        session.data example::
+
+        {
+          "interlocks_config": {
+            "configured": true,
+            "enabled": true,
+            "instance_id": "hwp-supervisor",
+            "limit_sun_avoidance": true,
+            "tolerance": 0.1,
+          },
+          "supervisor_data": {
+            "timestamp": 1744692973.185377,
+            "ok": true,
+            "err_msg": "",
+            "_grip_brakes": [1, 1, 1],
+            "_grip_state": "ungripped",
+            "_is_spinning": true,
+            "_target_freq": 2.1,
+            "grip_state": "ungripped",
+            "spin_state": "spinning",
+            "request_block_motion": null,
+            "request_block_motion_timestamp": null
+          },
+          "allowed": {
+            "el": [true, [40, 70], [
+              [40, 70],
+            ],
+            "az": [true, [40, 70], [
+              [40, 70],
+            ],
+            "third": [false, null, []],
+          }
+        }
+
+        """
+        if not self.hwp_rules.configured:
+            session.data = {
+                'interlocks_config': self.hwp_rules.encoded(basic=True),
+            }
+            return True, "HWP Interlocks not configured."
```

Comment:
Should this not return `False`?

Reply:
Yes, totally.  Changed (but had to mess with "startup" a little bit too).